### PR TITLE
#1 - Testing Repository with bug.

### PR DESCRIPTION
**Pull request by veeko21**

### CODE REVIEW

File: src/main/java/com/example/demo/student/Student.java [hunk 1]
```diff
@@ -21,23 +22,23 @@
     private String name;
     private String email;
     private LocalDate dob;
+
+    @Transient
     private int age;
```
Comment (emprestes) [inline src/main/java/com/example/demo/student/Student.java]: A transient age? Why not provide a method to calculate the age base on dob field?

File: src/main/java/com/example/demo/student/StudentService.java [hunk 2]
```diff
@@ -1,21 +1,35 @@
...
+    //instead of saying that we will return this list > return List.of();
     public List<Student> getStudents() {
-        return List.of(new Student(
-                1L,
-                "Mariam",
-                "mariam.jamal@gmail.com",
-                LocalDate.of(2000, Month.JANUARY, 5),
-                21
-        ));
+        return studentRepository.findAll();//this returns a list to us
```
Comment (emprestes) [inline src/main/java/com/example/demo/student/StudentService.java:27]: 👍🏽

**Review by emprestes — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/demo/student/StudentRepository.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import java.util.Optional;
+
+//we need to specify the type of object we want to work with and
+//id for the type that we want
+@Repository //this interface is responsible for data access
+public interface StudentRepository extends JpaRepository<Student, Long> {
+
+    //SELECT * FROM student WHERE email = ?
+    @Query("SELECT s FROM Student s WHERE s.email = ?1")
```
Comment (emprestes) [inline src/main/java/com/example/demo/student/StudentRepository.java:15]: `:email` should be better than `?1`

**Review by emprestes — CHANGES_REQUESTED** [review]

**Review by emprestes — APPROVED** [review]
Good job!

File: src/main/java/com/example/demo/student/Student.java (hunk 1 above)
Comment (emprestes) [reply src/main/java/com/example/demo/student/Student.java]: ` pubilc int getAge() { // logic to calculate the age here base on dob field } `

Comment (veeko21) [reply src/main/java/com/example/demo/student/Student.java]: `public int getAge() {
        return Period.between(this.dob, LocalDate.now()).getYears();
 }`

**Pull request merged**